# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/TextEditSubActionHandler.cpp
# Commit: 7b4beaa4fab1
# Full Hash: 7b4beaa4fab16011913fea3582f0d89f22890c78
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2025-03-13 10:29:34
# Regressor Bug: 1717155
# File Overlap Count: 1
# Description:
#   Bug 1717155 - part 1: Make `TextEditor` use `GetTextNode()` more r=m_kato
#   
#   The anonymous `<div>` of `TextEditor` has only one `Text` as its first child.
#   Therefore, we can make `TextEditor` simpler to access the `Text` node.
#   
# ==============================================================================

diff -r 267514795e90 -r 7b4beaa4fab1 editor/libeditor/TextEditSubActionHandler.cpp
--- a/editor/libeditor/TextEditSubActionHandler.cpp	Thu Mar 13 04:48:38 2025 +0000
+++ b/editor/libeditor/TextEditSubActionHandler.cpp	Thu Mar 13 05:25:45 2025 +0000
@@ -527,9 +527,7 @@
 
   MaybeDoAutoPasswordMasking();
 
-  RefPtr<Element> anonymousDivElement = GetRoot();
-  RefPtr<Text> textNode =
-      Text::FromNodeOrNull(anonymousDivElement->GetFirstChild());
+  const RefPtr<Text> textNode = GetTextNode();
   MOZ_ASSERT(textNode);
 
   // We can use this fast path only when:
@@ -624,8 +622,8 @@
         return Err(NS_ERROR_FAILURE);
       }
 
-      if (const Text* theTextNode = AsTextEditor()->GetTextNode()) {
-        rangesToDelete.EnsureRangesInTextNode(*theTextNode);
+      if (const Text* const textNode = GetTextNode()) {
+        rangesToDelete.EnsureRangesInTextNode(*textNode);
       }
 
       Result<CaretPoint, nsresult> caretPointOrError =
@@ -706,7 +704,7 @@
     return EditActionResult::HandledResult();
   }
 
-  Text* textNode = Text::FromNodeOrNull(anonymousDivElement->GetFirstChild());
+  const Text* const textNode = GetTextNode();
   MOZ_ASSERT(textNode);
 
   if (!textNode->Length()) {